### PR TITLE
use lld/mingw if explicitly selected via -mscrtlib=msvcrt100

### PR DESCRIPTION
**Pull request by rainers**
This should also help setting up some CI tests for this toolchain.

### CODE REVIEW

File: src/dmd/link.d [hunk 1]
```diff
@@ -251,7 +251,11 @@ public int runLINK()
             }
 
             VSOptions vsopt;
-            vsopt.initialize();
+            // if a runtime library (msvcrtNNN.lib) from the mingw folder is selected explicitly, do not detect VS and use lld
+            if (global.params.mscrtlib.length <= 6 ||
+                global.params.mscrtlib[0..6] != "msvcrt" || !isdigit(global.params.mscrtlib[6]))
```
Comment (thewilsonator) [inline src/dmd/link.d]: This look like it will match msvcrt869675436, not just msvcrt100

Comment (rainers) [reply src/dmd/link.d]: Yeah, I wanted to cover the versions that we might distribute once we upgrade to the mingw folders to what's build by https://github.com/dlang/installer/pull/346. LDC already has these, i.e. msvcrt80/90/90d/100/110/120/120d/120_app.

**Comment by Basile-z** [comment]
You need to make a dlang.org pull request too to update the command line reference (or update the place where it is generated), otherwise this will be forgotten fast.

**Comment by rainers** [comment]
> You need to make a dlang.org pull request

Command line documentation is generated from cli.d, I updated that.

File: src/dmd/link.d (hunk 1 above)
Comment (rainers) [reply src/dmd/link.d]: Should I defer that to when these are actually used?

Comment (wilzbach) [reply src/dmd/link.d]: Na I think it's more than fine to already check for it as it doesn't hurt at all and we will go this way.

**Review by ghost — APPROVED** [review]

File: src/dmd/link.d (hunk 1 above)
Comment (kinke) [reply src/dmd/link.d]: > I wanted to cover the versions that we might distribute

You may want to include a `vcruntime` prefix too then; `vcruntime140` is the UCRT/VC14 equivalent (unless you're planning to rename it).

**Review by wilzbach — APPROVED** [review]

File: src/dmd/link.d (hunk 1 above)
Comment (rainers) [reply src/dmd/link.d]: > You may want to include a `vcruntime` prefix too then; `vcruntime140` is the UCRT/VC14 equivalent (unless you're planning to rename it).

Not sure, we'll have to revisit this when we get to that. I'm also not happy with msvcrt.lib in the LDC folder,, as this is in conflict with the regular MS library.

**Pull request merged**